avm/res/document-db/database-account/secrets-key-vault/main.bicep

```
@@ -0,0 +1,34 @@
+metadata name = 'DocumentDB Database Account Key vault secrets'
+metadata description = 'This module deploys a the connection strings and keys into a key vault specified by the consumer'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. The key vault name where to store the keys and connection strings generated by the modules.')
+param keyVaultName string
+
+@description('Required. The name of the secret that will store the primary write key.')
+param keySecrets keySecret[]
+
+resource kv 'Microsoft.KeyVault/vaults@2022-07-01' existing = {
+  name: keyVaultName
+}
+
+resource keySecretsSecrets 'Microsoft.KeyVault/vaults/secrets@2022-07-01' = [for (secret, index) in keySecrets: {
+  name: secret.secretName
+  parent: kv
+  properties: {
+    value: secret.secretValue
+  }
+}]
+
+type keySecret = {
+  secretName: string
+  secretValue: string
```

```suggestion
  @secure()
  secretValue: string
```
I strongly suggest to make this parameter `secure()` to not expose the values in the deployment object in plain text.

---

Done as well